hail/python/hail/backend/backend.py

```diff
@@ -55,6 +56,11 @@ def matrix_type(self, mir):
         jir = self._to_java_ir(mir)
         return tmatrix._from_java(jir.typ())
 
+    def blockmatrix_type(self, bmir):
```

Comment:
You should implement blockmatrix_type on ServiceBackend, too.

Reply:
Added it.